vendor/k8s.io/kubernetes/staging/src/k8s.io/client-go/util/jsonpath/jsonpath.go

```diff
@@ -248,6 +248,11 @@ func (j *JSONPath) evalArray(input []reflect.Value, node *ArrayNode) ([]reflect.
 		}
 		if params[0].Value < 0 {
 			params[0].Value += value.Len()
+
+			// if value still less than zero, error out
+			if params[0].Value < 0 {
```

Comment:
Why would you do this?  If you've decided to tolerate some amount of negativity, is there some reason (spec-wise) to not continue the wrap around?